dask/tests/test_distributed.py

```diff
@@ -546,6 +546,39 @@ def f(d, a):
     await c.compute(df.map_partitions(f, arr, meta=df._meta))
 
 
+def test_map_partitions_df_input():
+    """
+    Check that map_partitions can handle a delayed
+    partition of a dataframe input
+    """
+    pd = pytest.importorskip("pandas")
+    dd = pytest.importorskip("dask.dataframe")
+    dist = pytest.importorskip("distributed")
+
+    def f(d, a):
+        assert isinstance(d, pd.DataFrame)
+        assert isinstance(a, pd.DataFrame)
+        return d
+
+    def main():
+        delayed_df = dd.from_pandas(
+            pd.DataFrame({"a": range(5)}), npartitions=2
+        ).to_delayed()
+        dl = delayed_df[0].persist()
+        wait(dl)
+
+        df = dd.from_pandas(pd.DataFrame({"a": range(5)}), npartitions=2)
+        df = df.map_partitions(f, dl, meta=df._meta)
+        df = df.persist(optimize_graph=False)
+        df.compute()
+
+    with dist.LocalCluster(
+        scheduler_port=0, asynchronous=False, n_workers=1, nthreads=1, processes=False
+    ) as cluster:
+        with dist.Client(cluster, asynchronous=False):
```

Comment:
Can we drop this section in favor of using `@gen_cluster` like most of the other tests in this module? `gen_cluster` tends to be both lighter on CI and makes for easier debugging

Reply:
For some reason using `@gen_cluster` doesn't trigger the bug in https://github.com/dask/distributed/issues/4980.

Reply:
I suspect that something more is wrong regarding Blockwise and delayed objects. This PR is the a first step but I will  investigate.